src/assets.zig

```diff
@@ -42,106 +108,126 @@ fn readDefaultFile(allocator: NeverFailingAllocator, dir: std.fs.Dir) !ZonElemen
+			// unix style.
+			for(0..directoryEntry.path.len - fileSuffixLen) |i| {
+				if(directoryEntry.path[i] == '\\') {
+					assetId[folderName.len+1+i] = '/';
+				} else {
+					assetId[folderName.len+1+i] = directoryEntry.path[i];
 				}
+			}
 
-				const string = dir.readFileAlloc(main.stackAllocator.allocator, entry.path, std.math.maxInt(usize)) catch |err| {
-					std.log.err("Could not open {s}/{s}: {s}", .{subPath, entry.path, @errorName(err)});
+			const fileContent = addonAssetsSubdirectory.readFileAlloc(
+				main.stackAllocator.allocator,
+				directoryEntry.path,
+				std.math.maxInt(usize)
+			) catch |e| {
+				std.log.err("Could not open {s}/{s}: {s}", .{assetSubdirectory, directoryEntry.path, @errorName(e)});
+				continue;
+			};
+			defer main.stackAllocator.free(fileContent);
+
+			const zon = ZonElement.parseFromString(externalAllocator, fileContent);
+
+			// If this is migrations file, we interrupt normal asset processing and store it in migrations hashmap.
+			if (std.ascii.eqlIgnoreCase(directoryEntry.basename, "_migrations.zig.zon")) {
+				if (migrations == null) {
+					std.log.err("Migrations not allowed for {s}", .{assetSubdirectory});
 					continue;
-				};
-				defer main.stackAllocator.free(string);
-
-				const zon = ZonElement.parseFromString(externalAllocator, string);
-				if (defaults) {
-					const path = entry.dir.realpathAlloc(main.stackAllocator.allocator, ".") catch unreachable;
-					defer main.stackAllocator.free(path);
+				}
+				(migrations orelse unreachable).put(assetId, zon) catch unreachable;
+				// This means that we skip default file reading and storing file content as normal asset.
+				continue;
+			}
 
-					const result = defaultMap.getOrPut(path) catch unreachable;
+			if (defaults) {
+				const path = directoryEntry.dir.realpathAlloc(main.stackAllocator.allocator, ".") catch unreachable;
+				defer main.stackAllocator.free(path);
 
-					if (!result.found_existing) {
-						result.key_ptr.* = defaultsArenaAllocator.dupe(u8, path);
-						const default: ZonElement = readDefaultFile(defaultsArenaAllocator, entry.dir) catch |err| blk: {
-							std.log.err("Failed to read default file: {s}", .{@errorName(err)});
-							break :blk .null;
-						};
+				const result = defaultMap.getOrPut(path) catch unreachable;
 
-						result.value_ptr.* = default;
-					}
+				if (!result.found_existing) {
+					result.key_ptr.* = defaultsAllocator.dupe(u8, path);
+					const default: ZonElement = readDefaultFile(defaultsAllocator, directoryEntry.dir) catch |err| blk: {
+						std.log.err("Failed to read default file: {s}", .{@errorName(err)});
+						break :blk .null;
+					};
 
-					zon.join(result.value_ptr.*);
+					result.value_ptr.* = default;
 				}
-				output.put(id, zon) catch unreachable;
+
+				zon.join(result.value_ptr.*);
 			}
+			output.put(assetId, zon) catch unreachable;
 		}
 	}
 }
-/// Reads text files recursively from all subfolders.
```

Comment:
Grepped it, I think it is dead code, hence I have removed it and nothing exploded.